File: dev/test_data/main/java/expected/implementation_specific/input/snippets/Verification/items_have_unique_labels.java
/**
* Check that {@link IItem#getLabel() label}'s of the items do not repeat.
* @param items the items to be checked
*/
public static boolean itemsHaveUniqueLabels(Iterable<? extends IItem> items) {
  Objects.requireNonNull(items);

  Set<String> labelSet = new HashSet<>();
  for (IItem item : items) {
    if (labelSet.contains(item.getLabel())) {
      return false;
    }
    labelSet.add(item.getLabel());
  }
  return true;
}

File: dev/test_data/main/cpp/expected/implementation_specific/input/snippets/verification/items_have_unique_labels.cpp
bool ItemsHaveUniqueLabels(
  const std::vector<
    std::shared_ptr<types::IItem>
  >& items
) {
  std::set<std::wstring> label_set;

  for (const std::shared_ptr<types::IItem>& item : items) {
    const std::wstring& label = item->label();

    if (label_set.find(label) != label_set.end()) {
      return false;
    }

    label_set.insert(label);
  }

  return true;
}
